native/golang/worker.go

```diff
@@ -218,6 +225,27 @@ func initializeWorker(worker *Worker, pool *WorkerPool) {
 		panic(err)
 	}
 
+	// Open the shared memory file
+	if worker.ShmPath != "" {
+		fd, err := syscall.Open(worker.ShmPath, syscall.O_RDWR, 0600)
+		if err != nil {
+			utils.LogFatal("Worker %d: Failed to open shared memory file %s: %v\n", worker.ID, worker.ShmPath, err)
+		} else {
+			worker.ShmFD = int(fd)
+		}
+	}
+
+	// Map the shared memory using the file descriptor
+	if worker.ShmFD > 0 {
+		// Use unix.Mmap for memory mapping
+		data, err := unix.Mmap(worker.ShmFD, 0, 68, unix.PROT_READ|unix.PROT_WRITE, unix.MAP_SHARED)
```

Comment:
The size passed by `unix.Mmap` should be 12 to match `sizeof(ZSharedRegion)`:

```suggestion
		data, err := unix.Mmap(worker.ShmFD, 0, 12, unix.PROT_READ|unix.PROT_WRITE, unix.MAP_SHARED)
```

Sorry this might have been merged in from the last PR, but its probably best to only allocate what we need.
It would be nice to make this dynamic at some point, but I don't want to add scope creep 😋